pkg/management/postgres/metrics/collector.go

```
@@ -69,13 +69,15 @@ var isPathPattern = regexp.MustCompile(`[][*?]`)
 
 // Update recomputes the metrics from the user queries
 func (q *QueriesCollector) Update() error {
+	isPrimary, err := q.instance.IsPrimary()
+	if err != nil {
+		return fmt.Errorf("while updating query metrics, could not check for primary: %w", err)
```

**Nit:** Should we explicitly state that the query is user provided/custom? (if we're not including a portion of the query to help identify it?) ... I'm thinking what if a user has multiple custom queries that are working, then they add one with a syntax error.... will it be obvious to know what's going wrong? [other than the absence of that query having results in the metrics?]

---

Now, reading between the lines, I think you're asking, what happens if there is an error when we're running the metric query `foo`.
And farther down in the code, in the `createMetricsFromUserQueries`, we have two ways we're recording this: into a metric collecting errors, and via a log
See lines 145-150
``` go
if err != nil {
				queryLogger.Error(err, "Error collecting user query",
					"targetDatabase", targetDatabase)
				// Increment metrics counters.
				q.reportUserQueryErrorMetric(name + " on db " + targetDatabase + ": " + err.Error())
			}
```

---

At the very bottom of this chain, we create a const metric or a const histogram.
And the error returned there generates a log, but the return is nil.
If that was your concern, then I do agree with it. Would probably be better to bubble the errors up.
That said, the existing behavior was this one (on error it logged but did not perform the "collection"), and I wanted to keep this PR from growing too much...

---

Yikes, sorry @jmealo , long answer but was not sure specifically what you were pointing out.
If you think worth doing the bubbling up of errors, I'd like the maintainers to take a look first, before the PR gets bigger. WDYT?